SpatialGDK/Source/SpatialGDKFunctionalTests/Private/GenerateTestMapsCommandlet.cpp

```diff
@@ -0,0 +1,276 @@
+void UGenerateTestMapsCommandlet::CreateSpatialComponentMap()
+{
+	UWorld* CurrentWorld = CreateNewTestMapWorld();
+	ULevel* CurrentLevel = CurrentWorld->GetCurrentLevel();
+
+	// Add the test
+	ASpatialComponentTest* CompTest = CastChecked<ASpatialComponentTest>(
+		GEditor->AddActor(CurrentLevel, ASpatialComponentTest::StaticClass(), FTransform(FVector(-250, -250, 0))));
+
+	// Add the helpers, as we need things placed in the level
+	CompTest->LevelActor = CastChecked<ASpatialComponentTestActor>(
+		GEditor->AddActor(CurrentLevel, ASpatialComponentTestActor::StaticClass(), FTransform(FVector(-250, -250, 0))));
+	CompTest->LevelReplicatedActor = CastChecked<ASpatialComponentTestReplicatedActor>(
+		GEditor->AddActor(CurrentLevel, ASpatialComponentTestReplicatedActor::StaticClass(), FTransform(FVector(-250, -250, 0))));
+
+	// Quirk of the test. We need the player spawns on the same portion of the map as the test, so they are LBed together
+	AActor** PlayerStart = CurrentLevel->Actors.FindByPredicate([](AActor* Actor) {
+		return Actor->GetClass() == APlayerStart::StaticClass();
+	});
+	(*PlayerStart)->SetActorLocation(FVector(-500, -250, 100));
+
+	ASpatialWorldSettings* WorldSettings = CastChecked<ASpatialWorldSettings>(CurrentWorld->GetWorldSettings());
+	WorldSettings->SetMultiWorkerSettingsClass(UTest1x2WorkerSettings::StaticClass());
+	WorldSettings->DefaultGameMode = ASpatialAuthorityTestGameMode::StaticClass();
+
+	FString PathToSave = FPaths::ConvertRelativePathToFull(FPaths::ProjectContentDir()
+														   + TEXT("Maps/FunctionalTests/GeneratedMaps/CI_Fast/SpatialComponentMap.umap"));
+	FEditorFileUtils::SaveLevel(CurrentLevel, PathToSave);
+}
+
+// Custom made for the SpatialTestHandover, placed in the slow category
+void UGenerateTestMapsCommandlet::CreateSpatialHandoverMap()
+{
+	UWorld* CurrentWorld = CreateNewTestMapWorld();
+	ULevel* CurrentLevel = CurrentWorld->GetCurrentLevel();
+
+	// Add the tests
+	// This test is here, because I cannot run it a 100 times in a row
+	GEditor->AddActor(CurrentLevel, ASpatialTestHandover::StaticClass(), FTransform::Identity);
+
+	ASpatialWorldSettings* WorldSettings = CastChecked<ASpatialWorldSettings>(CurrentWorld->GetWorldSettings());
+	WorldSettings->SetMultiWorkerSettingsClass(UTest2x2WorkerSettings::StaticClass());
+	WorldSettings->bEnableDebugInterface =
+		false; // The test directly accesses the LayeredStrategy and expects it to be the top one, setting to false for now
+
+	FString PathToSave = FPaths::ConvertRelativePathToFull(
+		FPaths::ProjectContentDir() + TEXT("Maps/FunctionalTests/GeneratedMaps/CI_Slow_Spatial_Only/SpatialHandoverMap.umap"));
+	FEditorFileUtils::SaveLevel(CurrentLevel, PathToSave);
+}
+
+int32 UGenerateTestMapsCommandlet::Main(const FString& CmdLineParams)
+{
+	UE_LOG(LogGenerateTestMapsCommandlet, Display, TEXT("Generate test maps commandlet started. Pray."));
+
+	UE_LOG(LogGenerateTestMapsCommandlet, Display, TEXT("Creating the SpatialNetworkingMap."));
```

Comment:
So one thing we could do to make this nicer, is to have a TestMap class, with each individual test map descending from it, e.g. `SpatialNetworkingMap : public TestMap`.
Then we could have member variables, shared behaviour, overrides, etc.
The tests could also auto-register themselves with the commandlet / the commandlet could search for all class descendants of the `TestMap` class.
Felt a bit complicated for a first pass, but could be nice.

Reply:
This would also allow us to nicely split this into multiple files, one for each test (that is possible with these functions too, I guess).

Reply:
I'm definitely more for this kind of design